### DISCUSSION TITLE
NFS Share Problem

### DISCUSSION THREAD

**Discussion by noz1380** [retronas/discussions#49] [Q&A]
When I try to access the NFS share on the RetroNAS, I can mount the first level IE storage, but I want to mount the next folder level up.  I hope this makes sense??

IE "my_Ip_Addy:/storage" mounts fine but "my_Ip_Addy:/storage/rgbpi-OS4" wont mount.

Can anyone assist please?

**Comment by sairuk** [comment]
to the best of my knowledge,

nfs mounts the export, you cannot mount folders under export unless you use something like `mount --bind` after mounting the export somewhere.

so to achieve what you are after you might mount storage somewhere other than the final mount dest then either symlink or `mount --bind` the rgbpi-OS4 folder to the final destination

alternatively you could provide a second export that targets the rgpbi-OS4 directory on the nfs server then mount that like any other

**Reply by noz1380** [reply]
Ok I think I follow.  Not used to NFS.

I'll try the last option when home this evening, see how I fair.

Thank you.

**Reply by noz1380** [reply]
![image](https://user-images.githubusercontent.com/63846328/181834036-c5300ead-d11a-48f1-98bd-d4d122814b91.png)

I think the problem is RGB-PI side....

**Reply by sairuk** [reply]
if you can get a terminal on the rgbpi side you should be able to do a couple of things to try and troubleshoot this

check dmesg output for errors
`# dmesg`

check the mount is visible for the rgbpi
`# showmount -e <retornas ip address>`

mount manually with verbosity and check for errors
`# mount -vvvv -t nfs  <retronas ip address>:/storage/rgbpi-OS4 <mount point>`

**Comment by noz1380** [comment]
I have used fstab in the past, but I'm unsure what this is asking....

![image](https://user-images.githubusercontent.com/63846328/181916567-222a8b1c-3da2-47d7-add2-33d0ba5b8a4c.png)

**Reply by sairuk** [reply]
command still needed a mount point, i edited my resp above to make it clearer sorry

**Comment by noz1380** [comment]
So, the work around, make a directory in the root, symlink to the other directory, then mount the root directory and add to exports.  All connects up now......RGB-PI OS should support sub-folders, but its at Alpha stage, and its not working correctly.

Only issue now is the the RGB-PI OS doent seem to like symlinks, and stops where I have nes symlink to /storage/roms/nintendo/famicom

I'll keep digging......
